include/seqan3/io/stream/debug_stream.hpp

```diff
@@ -295,6 +295,32 @@ inline debug_stream_type & operator<<(debug_stream_type & s, alphabet_t const l)
     return s << to_char(l);
 }
 
+/*!\brief A std::pair can be printed by printing the elements separately.
+ * \tparam pair_t1 Type of first element.
+ * \tparam pair_t2 Type of second element.
+ * \param s The seqan3::debug_stream.
+ * \param p The std::pair.
+ * \relates seqan3::debug_stream_type
+ */
+template <typename pair_t1, typename pair_t2>
+inline debug_stream_type & operator<<(debug_stream_type & s, std::pair<pair_t1, pair_t2> & p)
+//!\cond
+    requires requires (pair_t1 l) { { debug_stream << l }; } && requires (pair_t2 l) { { debug_stream << l }; }
+//!\endcond
+{
+    return s << '<' << p.first << ',' << p.second << '>';
+}
+
+//!\copydoc debug_stream_type & operator<<(debug_stream_type & s, std::pair<pair_t1, pair_t2> & p)
+template <typename pair_t1, typename pair_t2>
+inline debug_stream_type & operator<<(debug_stream_type & s, std::pair<pair_t1, pair_t2> const & p)
+//!\cond
+    requires requires (pair_t1 l) { { debug_stream << l }; } && requires (pair_t2 l) { { debug_stream << l }; }
```

Comment:
Is this a new style that we put the requires in the trailing location?

Reply:
What does trailing location mean in this context?

Reply:
Well, I particularluy find it less readable, but @h-2 seems to use it as it does not need to be guarded with a doxygen condition. But honestly I prefer the other way, because I see immediately the constraints on the types. But we can discuss this on the strategy meeting.

Reply:
For member functions of class templates (VS function templates) the trailing constraints notation is the only legal notation.
For function templates I also prefer the other notation, but if it means adding extra guards I usually switch to trailing notation, also because it means that the constraint is visible in doxygen.